### PR TITLE
Implement py-cache-decorator

### PR DESCRIPTION
**Pull request by SevKrok**

### CODE REVIEW

**Review by Sel-Fisher — APPROVED** [review]
i like it👍

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,14 @@
...
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def wrapper(*args) -> int:
+        if args in cached_results:
+            print("Getting from cache")
+        else:
```
Comment (fsocie7y) [inline app/main.py]: better try without else, use return in if instead

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,14 @@
...
+            cached_results[args] = func(*args)
+            print("Calculating new result")
+        return cached_results[args]
```
Comment (fsocie7y) [inline app/main.py:13]: then here do caching and return func result

**Review by fsocie7y — CHANGES_REQUESTED** [review]

**Review by skoval7 — APPROVED** [review]

**Comment by skoval7** [comment]
everything ok

**Review by LiudmylaKulinchenko — APPROVED** [review]
Well done 😋

**Pull request closed without merging**